Documentation/ceph-object-bucket-notifications.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Object Bucket Notification
+weight: 2950
+indent: true
+---
+
+# Ceph Object Bucket Notification
+
+Rook supports the creation of bucket notifications via two custom resources:
+
+- a `Ceph Bucket Topic (CBT)` is custom resource which represents a bucket notification topic and is described by a Custom Resource Definition (CRD) shown below. A bucket notification topic represents an endpoint (or a "topic" inside this endpoint) to which bucket notifications could be sent.
+- a `Ceph Bucket Notification (CBN)` is a custom resource the defines: topic, events and filters of a bucket notification, and is described by a CRD shown below. The bucket notification itself should also be associated with a bucket to complete its definition, however, this association is done by using special Object Bucket claim (OBC) labels, and is not part of the CBN.
+
+## Topics
+A CBT represents an endpoint (of types: Kafka, AMQP0.9.1 or HTTP), or a specific resource inside this endpoint (e.g a Kafka or an AMQP topic, or a specific URI in an HTTP server). The CBT also hold any additional info needed for the RADOS Gateway (RGW) to connect to that endpoint. Topics don't belong to a specific bucket or notification. Notifications from multiple buckets may be sent on the same topic, and one bucket (via multiple CBNs) may send notifications on multiple topics.
+
+## Notifications
+A CBN define: topic, events and filters of a bucket notifications. Meaning, it defines what triggers the notification, and where to send the notifications to. However, to complete the definition of the bucket notification, it must be associated with an OBC. This is done outside of the CBN resource, via adding labels to an OBC with the following format:
+```yaml
+bucket-notification-<notification name>: <notification name>
+```
+> Note that the CBT, CBN and OBC must all belong to the same namespace
+
+## Notification Reliability and Delivery
+Notifications may be sent synchronously, as part of the operation that triggered them. In this mode, the operation is acked only after the notification is sent to the topic’s configured endpoint, which means that the round trip time of the notification is added to the latency of the operation itself.
+> Note that the original triggering operation will still be considered as successful even if the notification fail with an error, cannot be delivered or times out
+
+Notifications may also be sent asynchronously. They will be committed into persistent storage and then asynchronously sent to the topic’s configured endpoint. In this case, the only latency added to the original operation is of committing the notification to persistent storage.
+> Note that if the notification fail with an error, cannot be delivered or times out, it will be retried until successfully acked
+
+## Sample
+
+### CBT Custom Resource
```

Comment:
Could we avoid the CBT and CBN acronyms? Informally we could refer just to topics and notifications, and when formally referring to the CR, we should spell it out for example to CephBucketTopic. Similarly we don't use acronyms for other CRDs such as CBP (CephBlockPool), CF (CephFilesystem), or COS (CephObjectStore). It's just too many acronyms to remember. :)  
```suggestion
### CephBucketTopic Custom Resource
```